internal/go_repository.bzl

```diff
@@ -260,6 +226,42 @@ go_repository = repository_rule(
 )
 """See repository.rst#go-repository for full documentation."""
 
+def get_env(ctx):
```

Comment:
Let's keep this code inlined in `_go_repository_impl`. I don't think it needs to be a separate function. It's fine to build the environment unconditionally.

Reply:
Done!